File: Behavioral/Memento/JavaScript/editor.js
// editor.js

const EditorMemento = require("./editorMemento");

class Editor {
  constructor() {
    this._content = "";
  }

  type(words) {
    this._content = words;
  }

  save() {
    return new EditorMemento(this._content);
  }

  restore(memento) {
    if (memento) {
      this._content = memento.getContent();
    }
  }

  getContent() {
    return this._content;
  }
}

module.exports = Editor;

File: Behavioral/Memento/CSharp/Editor.cs
// Editor.cs

public class Editor {
  private string content = "";

  public void Type(string words) {
    content = words;
  }

  public EditorMemento Save() {
    return new EditorMemento(content);
  }

  public void Restore(EditorMemento? memento) {
    if (memento != null) {
      content = memento.Content;
    }
  }

  public string GetContent() {
    return content;
  }
}
